internal/cephfs/nodeserver.go

```diff
@@ -94,31 +123,25 @@ func (ns *NodeServer) NodeStageVolume(
 	}
 	defer ns.VolumeLocks.Release(req.GetVolumeId())
 
-	volOptions, _, err := core.NewVolumeOptionsFromVolID(ctx, string(volID), req.GetVolumeContext(), req.GetSecrets())
+	volOptions, err := ns.getVolumeOptions(ctx, volID, req.GetVolumeContext(), req.GetSecrets())
 	if err != nil {
-		if !errors.Is(err, cerrors.ErrInvalidVolID) {
-			return nil, status.Error(codes.Internal, err.Error())
-		}
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+	defer volOptions.Destroy()
 
-		// gets mon IPs from the supplied cluster info
-		volOptions, _, err = core.NewVolumeOptionsFromStaticVolume(string(volID), req.GetVolumeContext())
-		if err != nil {
-			if !errors.Is(err, cerrors.ErrNonStaticVolume) {
-				return nil, status.Error(codes.Internal, err.Error())
-			}
+	m, err := mounter.New(volOptions)
+	if err != nil {
+		log.ErrorLog(ctx, "failed to create mounter for volume %s: %v", volID, err)
 
-			// get mon IPs from the volume context
-			volOptions, _, err = core.NewVolumeOptionsFromMonitorList(string(volID), req.GetVolumeContext(),
-				req.GetSecrets())
-			if err != nil {
-				return nil, status.Error(codes.Internal, err.Error())
-			}
-		}
+		return nil, status.Error(codes.Internal, err.Error())
 	}
-	defer volOptions.Destroy()
 
 	// Check if the volume is already mounted
 
+	if err = ns.tryRestoreFuseMountInNodeStage(ctx, m, stagingTargetPath); err != nil {
```

Comment:
@gman0 if the volume is already mounted on the nodes. kubelet will not issue the nodeStage again. as we are trying to restore here, when the Nodestage will be issued to correct this one? Only when the node is rebooted this will get triggered or even this can happen in other cases also?

Reply:
The general flow (iic ) goes like this: it  will try to fetch and restore if there is a record, otherwise skip and do the staging as in normal workflow.

Reply:
@Madhu-1

> kubelet will not issue the nodeStage again.

That's right, this is probably a dead code with kubelet at the moment. I've added it here just for the sake of "correctness": if NodeStage sees something is broken, it will try to fix it.

@humblec 

Workflow in `NodeStageVolume`:
* Is `staging_target_path` corrupted?
   * Yes: unmount `staging_target_path`.
   * No: fall through.
* Mount `staging_target_path`.

Workflow in `NodePublishVolume`:
* Is `staging_target_path` corrupted or unmounted, or `target_path` corrupted?
   * Yes: check for mountinfo record, and volume secrets. Try to remount `staging_target_path` and only unmount `target_path` (the bindmount will be restored in the next step)
   * No, or some of the prerequisites are not present: fall through.
* Is `staging_target_path` ok? If no, exit with error.
* Bindmount `staging_target_path` to `target_path`.